config/routes.rb

```diff
@@ -5,15 +5,26 @@
   # You can have the root of your site routed with "root"
   root 'task#index'
 
-  get "/task_:id/details" => "task#show"
+  post "/task_:id/mark_complete"   => "task#mark_complete"
+  post "/task_:id/mark_incomplete" =>      "task#mark_incomplete"
 
-  get "/task/new" => "task#new"
-  post "/task/new" => "task#create"
+  get "/task_:id/details"         => "task#show"
 
-  delete "/task_:id" => "task#delete"
+  get "/task/new"         => "task#new"
+  post "/task/new"        => "task#create"
+
+  delete "/task_:id"      => "task#delete"
+
+  post "/completed_sort"  => "task#sort_completed"
+  post "/name_sort"       => "task#sort_name"
+
+  get "/task_:id/edit"    => "task#edit"
+  patch "/task_:id/edit"  => "task#update"
+
+  get "/people"           => "people#index"
+
+  get "/assigned_to_:id" => "people#show"
```

Comment:
Do you think that the route would be easier to read if the `:id` were preceded by a `/`? For example `/assigned_to/1` rather than `/assigned_to_1`?

There's a lot of debate on how to build REST interfaces that filter some set of resources on a value, particularly an associated resource. Here are a couple of patterns I've seen that you could consider (following the plural-resource naming I described above):

```ruby
get "/tasks/assigned_to/:person_id" => "people#show"
get "/people/:id/tasks" => "people#show"
```

Or, as your controller action is just `people#show` what do you think about keeping the route simple and matching that:

```ruby
get "people/:person_id" => "people#show"
```